source/Missions/BoardGameLogic.cs

```diff
@@ -73,7 +75,7 @@ public void Dispose()
 
         private static readonly PropertyInfo OpposingAgentPropertyInfo = typeof(MissionBoardGameLogic).GetProperty("OpposingAgent");
 
-        public void StartGame(bool startFirst, Agent opposingAgent)
+        public async void StartGame(bool startFirst, Agent opposingAgent)
```

Comment:
async void is really bad and should be avoided at all costs.
Looks like the reason it's here is to call a `CheckOpponentConnected` which per comment below, doesn't need to be async, and to call `Task.Delay` which can probably be replaced by `Thread.Sleep`?